core/trino-main/src/main/java/io/trino/operator/TaskContext.java

```diff
@@ -349,6 +353,17 @@ public long getPhysicalWrittenDataSize()
         return physicalWrittenBytes;
     }
 
+    public void setMaxWriterCount(int maxWriterCount)
+    {
+        this.maxWriterCount.compareAndSet(0, maxWriterCount);
+    }
+
+    public Optional<Integer> getMaxWriterCount()
```

Comment:
nit: this could be evaluated within `getTaskStats`, no need for a public method

Reply:
This method is used inside `SqlTask#createTaskStatus`. Therefore, it has to be public.